diff --git a/test/regression_tests/test_utils.py b/test/regression_tests/test_utils.py
index 97f69ba..216d735 100644
--- a/test/regression_tests/test_utils.py
+++ b/test/regression_tests/test_utils.py
@@ -80,5 +80,6 @@ def test_interpolate_dynamics():
         )
     t3 = datetime.datetime.now()
 
-    assert (res1 == res2).all()
+    # comparing with tolerance when MKL is used
+    np.testing.assert_allclose(res1, res2, rtol=1e-12)
     assert (t3 - t2).total_seconds() <= 0.7 * (t2 - t1).total_seconds()
